src/Microsoft.OData.Edm/Vocabularies/Annotations/IEdmTerm.cs

```diff
@@ -25,5 +25,10 @@ public interface IEdmTerm : IEdmSchemaElement
         /// Gets the DefaultValue of this term.
         /// </summary>
         string DefaultValue { get; }
+
+        /// <summary>
+        /// Gets the base term of this term.
+        /// </summary>
+        IEdmTerm BaseTerm { get; }

```

Comment:
This is my only real concern with the PR.  We need to convince ourselves that we are the only ones implementing this interface, as anyone else providign a custom implementation will break when they update to the new library and have to modify their code to implement this property.

The fallback would be to define an IEdmTermWithBase : IEdmTerm, but that would require logic each place we consume IEdmTerm that we care about base to handle base as well, if the instance of IEdmTerm can be cast to IEdmTermWithBase.  We could add IEdmTermWithBase in 7.9 and clean it up in 8.0, or we can convince ourselves that there aren't any custom implementations of IEdmTerm that we would break.

Reply:
That's also my concern. However, i have the following two reasons:

1) To add new into a public interface is less "harmful" than to remove or to change element in the public interface.
2) I can't find any reason why there's a customer who need to implement this interface.



---
In reply to: [545401071](https://github.com/OData/odata.net/pull/1962#discussion_r545401071) [](ancestors = 545401071)

Reply:
I think I'm understand the high-level concern but missing the specifics for our case. Could we not have this property be of type `BaseTerm : IEdmTerm` that has an `internal` constructor so we have full control over the behavior?